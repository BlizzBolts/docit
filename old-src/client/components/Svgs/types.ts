import type { SVGAttributes } from "react";
import type React from "react";

export interface SvgProps extends SVGAttributes<SVGElement> {
  className?: string;
  style?: React.CSSProperties;
}
