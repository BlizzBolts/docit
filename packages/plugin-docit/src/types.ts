export interface DocitPluginOption {
  root: string;
}
