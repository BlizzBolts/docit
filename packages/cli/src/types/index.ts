export type ConfigFileLocation = string;

export interface CLIOptions {
  config?: ConfigFileLocation;
}
