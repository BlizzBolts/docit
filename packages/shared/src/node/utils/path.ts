import path from "node:path";
import url from "node:url";
/**
 * isomorphic __dirname getter
 *
 * @param {import.meta.url} importMetaUrl
 * @returns
 */
export const getDirname = (importMetaUrl: string) => {
  return path.dirname(url.fileURLToPath(importMetaUrl));
};
