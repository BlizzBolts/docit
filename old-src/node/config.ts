import path from "path";
import { isFileExists } from "./utils/index.js";
import type { UserConfig, ResolvedUserConfig } from "./types.js";
import { BUILD_DIST_PATH } from "./constants.js";

export const resolveConfig = async (
  config: UserConfig,
  command: "build" | "start"
): Promise<ResolvedUserConfig> => {
  return {
    base: path.resolve(BUILD_DIST_PATH, "./client"),
    docs: path.resolve(process.cwd(), config.root),
    title: config.title || "Docit",
    sidebars: config.sidebars || [],
    providerPath: isFileExists(config.providerPath)
      ? config.providerPath
      : path.resolve(
          BUILD_DIST_PATH,
          "./client/components/DefaultProvider/index.js"
        ),
    publicPath: command === "build" ? config.publicPath : "/",
    socials: config.socials,
    vite: config.vite,
  };
};
