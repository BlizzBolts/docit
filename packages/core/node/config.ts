import path from "node:path";
import { glob } from "glob";
import { bundleRequire } from "bundle-require";
import type { DocitConfig } from "@blizzbolts/docit-shared";
import { coreLogger, zDocitConfig, zPrintErr } from "@blizzbolts/docit-shared";

export interface ConfigFromFile {}

const readConfig = async () => {};

export const findConfigFile = async (
  cwd: string = process.cwd(),
  options?: {
    isEsm?: boolean;
  },
): Promise<string | null> => {
  const globString = options?.isEsm ? "./docit.config.{js,cjs,ts}" : "./docit.config.{mjs,js,ts}";

  const matches = await glob(globString, {
    cwd,
    nodir: true,
  });
  if (matches.length === 0) {
    return null;
  } else {
    return matches[0];
  }
};

export const readConfigFromFile = async (
  cwd: string = process.cwd(),
  options?: {
    isEsm?: boolean;
  },
): Promise<DocitConfig> => {
  const configFile = await findConfigFile(cwd, options);
  coreLogger.info("Locate docit.config file at", configFile);
  if (!configFile) {
    return {};
  }

  try {
    const { mod } = await bundleRequire({
      cwd,
      filepath: path.resolve(cwd, "./", configFile),
      getOutputFile: (filePath) => {
        return path.resolve(cwd, filePath);
      },
      tsconfig: "./tsconfig.json",
    });
    // I mean ... it works in both .cjs and .mjs situations
    return zDocitConfig.parse(mod.default ? mod.default : mod);
  } catch (e) {
    coreLogger.warn(`Failed to load config at ${configFile}, resolved as default\n`, e);
    zPrintErr(e);
    return {};
  }
};

const readConfigFromPackageJson = () => {};

const mergeConfig = () => {};

export const defaultConfig = {};

export const resolveConfig = () => {
  const config = readConfig();

  return config;
};
