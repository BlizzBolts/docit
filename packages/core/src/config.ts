import path from "node:path";
import { glob } from "glob";
import { bundleRequire } from "bundle-require";
import { coreLogger } from "@blizzbolts/docit-shared";

export interface Config {}

export interface ConfigFromFile {}

const readConfig = async () => {
  const configFromFile = await readConfigFromFile();

  return configFromFile;
};

export const readConfigFromFile = async (cwd: string = process.cwd()) => {
  const matches = await glob("./.docit/docit.config.{js,mjs,cjs,ts,mts,cts}", {
    cwd,
  });
  if (matches.length === 0) {
    return {};
  }
  try {
    const { mod } = await bundleRequire({
      cwd,
      filepath: matches[0],
      getOutputFile: (filePath) => {
        return path.resolve(cwd, filePath);
      },
    });
    return mod.default;
  } catch (e) {
    coreLogger.warn(`Failed to load config at ${matches[0]}, resolve as {}\n`, e);
    return {};
  }
};

const readConfigFromPackageJson = () => {};

const mergeConfig = () => {};

export const defaultConfig = {};

export const resolveConfig = () => {
  const config = readConfig();

  return config;
};
