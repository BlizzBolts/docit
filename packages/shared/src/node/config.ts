import path from "node:path";
import { glob } from "glob";
import { bundleRequire } from "bundle-require";
import type { DocitConfig } from "../shared/zod";
import { zDocitConfig } from "../shared/zod";
import { zPrintErr } from "../shared/zod";
import { coreLogger } from "../shared/logger";
import { isFileReadable } from "../node/utils/files";

export const findConfigFile = async (cwd: string = process.cwd()): Promise<string | null> => {
  // FIXME: ts config file support
  const globString = "./docit.config.{mjs,cjs,js}";

  const matches = await glob(globString, {
    cwd,
    nodir: true,
  });
  if (matches.length === 0) {
    coreLogger.info("Docit config not found, fallback to default");
    return null;
  } else {
    return matches[0];
  }
};

export const readConfigFromFile = async (
  cwd: string = process.cwd(),
): Promise<DocitConfig | null> => {
  const configFile = await findConfigFile(cwd);

  if (!configFile) {
    return null;
  }

  const isValidFile = isFileReadable(path.resolve(cwd, "./", configFile));

  if (!isValidFile) {
    return null;
  }
  try {
    const { mod } = await bundleRequire({
      filepath: path.resolve(cwd, "./", configFile),
    });
    // I mean ... it works in both .cjs and .mjs situations
    return mod.default ? mod.default : mod;
  } catch (e) {
    coreLogger.warn(`Failed to load config at ${configFile}, resolved as default\n`, e);
    zPrintErr(e);
    return null;
  }
};

const readConfigFromPackageJson = async (cwd: string = process.cwd()) => {
  return null;
};

const readConfig = async (cwd: string = process.cwd()) => {
  let config = null;
  const hasConfigFile = await findConfigFile(cwd);

  if (hasConfigFile) {
    config = await readConfigFromFile(cwd);
  } else {
    config = await readConfigFromPackageJson(cwd);
  }
  return config;
};

export const resolveConfig = async (cwd: string = process.cwd()): Promise<DocitConfig> => {
  cwd = path.resolve(cwd);
  const config = await readConfig(cwd);
  const resolvedConfig: DocitConfig = zDocitConfig.parse(config || {});

  if (resolvedConfig?.docRoot) {
    resolvedConfig.docRoot = path.resolve(cwd, resolvedConfig?.docRoot);
  }

  if (resolvedConfig?.outDir) {
    resolvedConfig.outDir = path.resolve(cwd, resolvedConfig?.outDir);
  }

  if (resolvedConfig?.root) {
    resolvedConfig.root = path.resolve(cwd, resolvedConfig.root);
  }

  resolvedConfig.server = {
    port: 3000,
    ...(resolvedConfig.server || {}),
  };

  return resolvedConfig;
};
