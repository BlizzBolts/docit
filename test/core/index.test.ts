import { afterEach, beforeEach, describe, expect, it } from "vitest";
import { Core } from "../../src/node/core/index.js";
import { resolveConfig } from "../../src/node/config.js";
import path from "path";
import { ResolvedUserConfig } from "../../src/node/types.js";

const makeFakeResolvedConfig = (): Promise<ResolvedUserConfig> => {
  return resolveConfig(
    {
      root: path.resolve(process.cwd(), "./docs"),
    },
    "start"
  );
};

describe("core test suite", () => {
  let instance: Core;

  beforeEach(async () => {
    instance = Core.getInstance(await makeFakeResolvedConfig());
    await instance.prepare();
  });

  afterEach(() => {
    instance = null;
  });

  it("Core prepared", async (complete) => {
    console.log(instance.getMarkdowns().length);
    expect(instance.getMarkdowns().length).not.toBe(0);
    complete();
  });

  // it("makes routes correctly", async (complete) => {
  //   expect(instance.makeRoutes()).toMatchSnapshot();
  //   complete();
  // });

  // it("makes sidebars correctly", async (complete) => {
  //   expect(await instance.makeSidebars()).toMatchSnapshot();
  //   complete();
  // });
});
