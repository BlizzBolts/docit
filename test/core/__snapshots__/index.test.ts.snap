// Vitest Snapshot v1

exports[`core test suite > makes routes correctly 1`] = `
"
      import React from 'react';

      export const routes = [
        
          {
            path: \\"/FAQ\\",
            exact: true,
            component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/docs/FAQ.mdx')),
          }
        ,
          {
            path: \\"/feature-request\\",
            exact: true,
            component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/docs/feature-request.mdx')),
          }
        ,
          {
            path: \\"/index\\",
            exact: true,
            component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/docs/index.mdx')),
          }
        ,
          {
            path: \\"/Test/GFM\\",
            exact: true,
            component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/docs/Test/GFM.md')),
          }
        ,
          {
            path: \\"/Test/demo\\",
            exact: true,
            component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/docs/Test/demo.mdx')),
          }
        ,
          {
            path: \\"/settings/index\\",
            exact: true,
            component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/docs/settings/index.mdx')),
          }
        ,
          {
            path: \\"/settings/sidebar\\",
            exact: true,
            component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/docs/settings/sidebar.mdx')),
          }
        ,
          {
            path: \\"/settings/static-resources\\",
            exact: true,
            component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/docs/settings/static-resources.mdx')),
          }
        ,
          {
            path: \\"/tutorials/GFM\\",
            exact: true,
            component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/docs/tutorials/GFM.mdx')),
          }
        ,
          {
            path: \\"/tutorials/MDX\\",
            exact: true,
            component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/docs/tutorials/MDX.mdx')),
          }
        ,
          {
            path: \\"/tutorials/using-components-in-docgen\\",
            exact: true,
            component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/docs/tutorials/using-components-in-docgen.mdx')),
          }
        ,
            {
              path: \\"*\\",
              exact: true,
              component: React.lazy(() => import('/Users/hao/local/projj/gitlab.alibaba-inc.com/cnfe-framework/docgen/src/client/404.js')),
            }
          
      ]
    "
`;

exports[`core test suite > makes sidebars correctly 1`] = `
[
  {
    "path": "/FAQ",
    "title": "FAQ",
  },
  {
    "path": "/feature-request",
    "title": "Feature Request",
  },
  {
    "path": "/index",
    "title": "介绍",
  },
  {
    "children": [
      {
        "path": "/Test/GFM",
        "title": "Editor.md",
      },
      {
        "path": "/Test/demo",
        "title": "Demo23345",
      },
    ],
    "title": "Test",
  },
  {
    "children": [
      {
        "path": "/settings/index",
        "title": "页面配置",
      },
      {
        "path": "/settings/sidebar",
        "title": "侧边栏",
      },
      {
        "path": "/settings/static-resources",
        "title": "静态资源",
      },
    ],
    "title": "settings",
  },
  {
    "children": [
      {
        "path": "/tutorials/GFM",
        "title": "GFM",
      },
      {
        "path": "/tutorials/MDX",
        "title": "MDX",
      },
      {
        "path": "/tutorials/using-components-in-docgen",
        "title": "如何书写使用 Components",
      },
    ],
    "title": "tutorials",
  },
]
`;
